### ISSUE TITLE
Cannot replace edge when there are more than one graph

### ISSUE THREAD

**Issue by ysmu** [arangodb/arangodb#10339]
## My Environment

* __ArangoDB Version__:        3.5.1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               Nothing special
* __Infrastructure__:               own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:        16Gb
* __Disks in use__:        SSD
* __Used Package__:              Docker - official Docker library

## Steps to reproduce

1. Create graph `graph_a`.
```sh
curl -X POST --header 'accept: application/json' --data-binary @- --dump - http://localhost:8529/_api/gharial <<EOF
{ 
  "name" : "graph_a"
}
EOF
```
```
HTTP/1.1 202 Accepted
```
2. Add an edge definition to the graph.
```sh
curl -X POST --header 'accept: application/json' --data-binary @- --dump - http://localhost:8529/_api/gharial/graph_a/edge <<EOF
{ 
  "collection" : "edge_a", 
  "from" : [ 
    "col_a"
  ], 
  "to" : [ 
    "col_a"
  ] 
}
EOF
```
```
HTTP/1.1 202 Accepted
```
3. Replace the edge definition with the same edge definition (really a no-op). This is important for later.
```sh
curl -X PUT --header 'accept: application/json' --data-binary @- --dump - http://localhost:8529/_api/gharial/graph_a/edge/edge_a <<EOF
{ 
  "collection" : "edge_a", 
  "from" : [ 
    "col_a"
  ], 
  "to" : [ 
    "col_a"
  ] 
}
EOF
```
```
HTTP/1.1 202 Accepted
```
4. Now, create another graph `graph_b`.
```sh
curl -X POST --header 'accept: application/json' --data-binary @- --dump - http://localhost:8529/_api/gharial <<EOF
{ 
  "name" : "graph_b"
}
EOF
```
```
HTTP/1.1 202 Accepted
```
5. Rerun the same command in 3) to replace the edge definition in `graph_a`. I was expecting this to succeed.
```sh
curl -X PUT --header 'accept: application/json' --data-binary @- --dump - http://localhost:8529/_api/gharial/graph_a/edge/edge_a <<EOF
> {
>   "collection" : "edge_a",
>   "from" : [
>     "col_a"
>   ],
>   "to" : [
>     "col_a"
>   ]
> }
> EOF
```
```
HTTP/1.1 400 Bad Request
X-Content-Type-Options: nosniff
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 117

{"code":400,"error":true,"errorMessage":"edge collection does not exist or is not part of the graph","errorNum":1939}
```
6. Sanity check to ensure the edge collection is in `graph_a`.
```sh
curl --header 'accept: application/json' --dump - http://localhost:8529/_api/gharial/graph_a/edge
```
```
HTTP/1.1 200 OK
X-Content-Type-Options: nosniff
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 51

{"error":false,"code":200,"collections":["edge_a"]}
```

__Problem__ & __Expected Results__: I was expecting 5) to return the same result as 3). The edge definition does exist in the graph as shown in 6), and it worked before I created a second graph in 4).

**Comment by OmarAyo** [comment]
Hi @ysmu 

We reproduce this, we will discuss internally and get back 

Best

**Comment by ysmu** [comment]
Any updates?

**Comment by maxkernbach** [comment]
Hi @ysmu,

Thanks for reporting this bug, a fix (https://github.com/arangodb/arangodb/pull/10690) will be available in the upcoming versions 3.5.4 and 3.4.9.
From the Changelog:
> * Fixed internal issue #4748: Editing a single edgeDefinition using the graph
  API failed if it was not shared between all available graphs.

**Comment by ysmu** [comment]
Thanks for the fix!
